### ISSUE TITLE
[nuclei-template] Exposed .gitignore

### ISSUE THREAD

**Issue by thezakman** [projectdiscovery/nuclei-templates#1898]
**Exposed .gitignore - sensitive resource**

```yaml
id: exposed-gitignore

info:
  name: Exposed .gitignore
  author: TheZakMan
  severity: info
  description: Expose sensitive resources 
  tags: config,git,exposure

requests:
  - method: GET
    path:
      - "{{BaseURL}}/.gitignore"
      - "{{BaseURL}}/assets/.gitignore"
      - "{{BaseURL}}/includes/.gitignore"

    matchers-condition: and
    matchers:
      - type: status
        status:
          - 200

      - type: dsl
        dsl:
          - '!contains(body_2, "<html")'
          - '!contains(body_2, "<HTML")'
        condition: and
```

**Comment by thezakman** [comment]
Forgot the reference:  https://twitter.com/pratiky9967/status/1230001391701086208

**Comment by ehsandeep** [comment]
Hi @thezakman, Check this out https://githubapp.com/.gitignore, it will be matched with the above template but not a valid match, I'm not sure without a unique body string (which looks not an option in this case) it's possible to match for valid cases only, what you think?

**Comment by thezakman** [comment]
I think like many templates, there will be false positives. But with time and the help of the community we can fine tune it to get better and more accurate over time, don't you think?

I got lots of false positives with some other templates, since this case it's a file that doesn't follow a certain pattern, I think more filter can be added, but the certainty will only be sure checking out the content yourself. No?

**Comment by ehsandeep** [comment]
@thezakman, 

> I got lots of false positives with some other templates

Please if it's possible let us know so we can improve/fix/remove those templates, but it's not intended to have a template with known false-positive possibilities.

**Comment by ehsandeep** [comment]
I've added content-type check to avoid false positives, but I'm sure this might miss some true positives as well, we can keep this one till the time we learn about improved matcher.

**Comment by thezakman** [comment]
@ehsandeep Don't you think closing this so fast, will not loose the opportunity of someone else come up with a clever idea, to improve the detection?

Like I mentioned, some others CVEs templates or miscellaneous ones, dosen't always hit the target that were supposed to. But that is ok, since every server has a different setup and config. That sometimes will send a false results.

Anyway, thanks for the quick reply.

**Comment by geeknik** [comment]
This additional matcher might be helpful:
```
- type: dsl
  dsl:
    - "len(body) > 50 "
```

**Comment by ehsandeep** [comment]
@thezakman

> Don't you think closing this so fast, will not loose the opportunity of someone else come up with a clever idea, to improve the detection?

Looks like we misunderstood something, we merged this template here https://github.com/projectdiscovery/nuclei-templates/pull/1913, as a result, GitHub auto closes linked issues.

**Comment by thezakman** [comment]
I totally missundertood it, sorry for any trouble.

And thanks once again @ehsandeep and @geeknik you guys rock!

**Comment by ehsandeep** [comment]
all good as you understand now 😄 @geeknik updated the template with an additional matcher.

**Comment by thezakman** [comment]
I think it would work better if this line is removed, no?

      - type: dsl
        dsl:
          - 'contains(tolower(all_headers), "binary/octet-stream")'


I did some tests and was not able to find anything with this line, after removal it worked like a charm.

**Comment by thezakman** [comment]
id: exposed-gitignore

info:
  name: Exposed Gitignore
  author: TheZakMan
  severity: info
  tags: config,git,exposure
  reference: |
      - https://twitter.com/pratiky9967/status/1230001391701086208
      - https://www.tenable.com/plugins/was/98595
requests:
  - method: GET
    path:
      - "{{BaseURL}}/.gitignore"
      - "{{BaseURL}}/assets/.gitignore"
      - "{{BaseURL}}/includes/.gitignore"

    matchers-condition: and
    matchers:
      - type: status
        status:
          - 200

      - type: dsl
        dsl:
          - '!contains(tolower(body), "<html") && !contains(tolower(body), "<!doctype") &&  !contains(tolower(body), "<script")'

      - type: dsl
        dsl:
          - "len(body) > 50 "

**Comment by geeknik** [comment]
> I think it would work better if this line is removed, no?
> 
> ```
>   - type: dsl
>     dsl:
>       - 'contains(tolower(all_headers), "binary/octet-stream")'
> ```
> 
> I did some tests and was not able to find anything with this line, after removal it worked like a charm.

Fixed by #2178
